TA_dataset/bin/dataset_common.py

```diff
@@ -224,6 +320,11 @@ def get_token(self, account, rw, logger):
                 and credential.username.startswith(account)
             ):
                 cred = credential.content.get("clear_password")
+                if rw == "authn":
+                    if key_field in cred:
+                        logger.info("the yes on authn token")
+                        cred_json = json.loads(cred)
+                        token = cred_json[key_field]
                 if rw == "read":
```

Comment:
consider chaning value `read` and `write` to apitoken_read/write so its more explicit

Reply:
we do supporting the read and write access SDL token as well, that the reason we don't changed the existing ones

Reply:
Do you mean this value is stored in Splunk config? So in order to not break backwards compatibility its not possible?

Reply:
correct in-order to achieve the backward compatibility

Reply:
this is just for identifying the type of the key, read/write

Reply:
Yes and that was my point that i would be better to change this identifier in order to improve readability